bluesky/protocols.py

```diff
@@ -1,17 +1,39 @@
 try:
-    from typing import Protocol, runtime_checkable
+    from typing import Literal, Protocol, TypedDict, runtime_checkable
 except ImportError:
-    from typing_extensions import Protocol, runtime_checkable
+    from typing_extensions import Literal, Protocol, TypedDict, runtime_checkable
 
-from typing import Dict, Any, Optional, Callable, Generator, List, Union, NoReturn
+from typing import Any, Awaitable, Callable, Dict, Generator, List, Optional, TypeVar, Union
 
 
-Configuration = Dict[str, Dict[str, Any]]
+class EventData(TypedDict):
+    value: Any
+    timestamp: float
+    # TODO: what about status/severity?
+
+
+class EventDescriptorMandatory(TypedDict):
+    source: str
+    dtype: Literal["string", "number", "array", "boolean", "integer"]
+    shape: List[int]
+
+
+class EventDescriptor(EventDescriptorMandatory, total=False):
```

Comment:
As per https://blueskyproject.io/bluesky/hardware.html#bluesky.protocols.Readable.describe

Are there more things that we can add here?

Reply:
yes